Foundation/NSString.swift

```diff
@@ -405,15 +405,39 @@ extension NSString {
     }
     
     public func rangeOfCharacterFromSet(searchSet: NSCharacterSet) -> NSRange {
-        NSUnimplemented()
+        return rangeOfCharacterFromSet(searchSet, options: [])
     }
     
     public func rangeOfCharacterFromSet(searchSet: NSCharacterSet, options mask: NSStringCompareOptions) -> NSRange {
-        NSUnimplemented()
+        return rangeOfCharacterFromSet(searchSet, options: mask, range: NSMakeRange(0, length))
     }
     
     public func rangeOfCharacterFromSet(searchSet: NSCharacterSet, options mask: NSStringCompareOptions, range searchRange: NSRange) -> NSRange {
-        NSUnimplemented()
+        if mask.contains(.RegularExpressionSearch) {
+            NSUnimplemented()
+        }
+        if searchRange.length == 0 {
+            return NSMakeRange(NSNotFound, 0)
+        }
+        
+#if os(Linux)
+        var cfflags = CFStringCompareFlags(mask.rawValue)
+        if mask.contains(.LiteralSearch) {
+            cfflags |= UInt(kCFCompareNonliteral)
+        }
+#else
+        var cfflags = CFStringCompareFlags(rawValue: mask.rawValue)
+        if mask.contains(.LiteralSearch) {
+            cfflags.unionInPlace(.CompareNonliteral)
+        }
+#endif
+        var result = CFRangeMake(kCFNotFound, 0)
+
```

Comment:
there probably should be a precondition here but overall the implementation looks good to me.